aiida/tools/importexport/dbexport/__init__.py

```diff
@@ -108,28 +138,38 @@ def export(
         )
 
     # Backwards-compatibility
-    entities = deprecated_parameters(
-        old={
-            'name': 'what',
-            'value': kwargs.pop('what', None)
-        },
-        new={
-            'name': 'entities',
-            'value': entities
-        },
+    entities = cast(
```

Comment:
just wondering - wouldn't it be easier to annotate the `deprecated_parameters` function instead?

Reply:
typing of variables was only introduced in python 3.6. I can't remember are we allowed to drop 3.5 now?

Reply:
yes, see https://github.com/aiidateam/aiida-core/pull/4386
next aiida-core release will be 1.5.0 without python 3.5 support

Reply:
ah actually you still need to cast here, because the function signature says that `entities` can also be `None`, then this cast "updates" that to say it is now only an iterable

Reply:
also `deprecated_functions` can return multiple different types